src/main/java/slimeknights/tconstruct/library/recipe/tinkerstation/building/ToolBuildingRecipe.java

```diff
@@ -48,13 +52,44 @@ public class ToolBuildingRecipe implements ITinkerStationRecipe {
   @Getter
   protected final IModifiable output;
   protected final int outputCount;
+  @Nullable
+  protected ResourceLocation layoutSlot;
   protected final List<Ingredient> ingredients;
 
+  /** @deprecated Use {@link #ToolBuildingRecipe(ResourceLocation, String, IModifiable, int, ResourceLocation, List)} */
+  @Deprecated
+  public ToolBuildingRecipe(ResourceLocation id, String group, IModifiable output, int outputCount, List<Ingredient> ingredients) {
+    this(id, group, output, outputCount, null, ingredients);
+  }
+
+  public ToolBuildingRecipe(ResourceLocation id, String group, IModifiable output, int outputCount, ResourceLocation layoutSlot, List<Ingredient> ingredients) {
+    this.id = id;
+    this.group = group;
+    this.output = output;
+    this.outputCount = outputCount;
+    this.layoutSlot = layoutSlot;
+    this.ingredients = ingredients;
+  }
+
+  public ResourceLocation getLayoutSlotId() {
+    return Objects.requireNonNullElse(layoutSlot, getOutput().getToolDefinition().getId());
+  }
+
+  public List<IToolPart> getToolParts() { return ToolPartsHook.parts(getOutput().getToolDefinition()); }
```

Comment:
Toss some quick function javadoc comments on these explaining what they do, see `getExtraRequirements` for an example. My rule of thumb is every method must be either `@Override` or have a javadoc comment. 

Also, please don't make a function in just 1 line, write it out over 3 at minimum.